api_tutorials/bait_cnn/bait_performance_cnn.py

```diff
@@ -0,0 +1,605 @@
+            total_baits += 1
+
+            if contig not in bed_with_cov_and_gc.keys():
+                bed_with_cov_and_gc[contig] = ([], [], [], [],)
+
+            bed_with_cov_and_gc[contig][0].append(lower)
+            bed_with_cov_and_gc[contig][1].append(upper)
+            bed_with_cov_and_gc[contig][2].append(reads)
+            bed_with_cov_and_gc[contig][3].append(gc)
+
+    reads_per_bait = float(total_reads) / total_baits
+    print("Reads per bait: %s" % reads_per_bait)
+    for contig in bed_with_cov_and_gc.keys():
+        bed_with_cov_and_gc[contig] = (
+            np.array(bed_with_cov_and_gc[contig][0]),
+            np.array(bed_with_cov_and_gc[contig][1]),
+            np.array([x / reads_per_bait for x in bed_with_cov_and_gc[contig][2]]),
+            bed_with_cov_and_gc[contig][3],)
+        print('key is: %s len %s' % (contig, len(bed_with_cov_and_gc[contig][0])))
+
+    return bed_with_cov_and_gc
+
+
+def in_bed_file(bed_dict, contig, pos):
+    lows = bed_dict[contig][0]
+    ups = bed_dict[contig][1]
+
+    return np.any((lows <= pos) & (pos <= ups))
+
+
+def split_data(data, valid_ratio=0.1, test_ratio=0.4):
+    # type: (Data, float, float) -> (Data, Data, Data)
+
+    np.random.seed(0)
+
+    samples = data.samples
+    indices = range(samples)
+    np.random.shuffle(indices)
+
+    valid_idx = int(valid_ratio * float(samples))
+    test_idx = int(test_ratio * float(samples))
+
+    train = Data(samples - valid_idx - test_idx, data.window, data.depth)
+    valid = Data(valid_idx, data.window, data.depth)
+    test = Data(test_idx, data.window, data.depth)
+
+    valid.coverage = data.coverage[:valid_idx]
+    valid.baitdata = data.baitdata[:valid_idx]
+    valid.gc = data.gc[:valid_idx]
+
+    test.coverage = data.coverage[valid_idx:(valid_idx + test_idx)]
+    test.baitdata = data.baitdata[valid_idx:(valid_idx + test_idx)]
+    test.gc = data.gc[valid_idx:(valid_idx + test_idx)]
+
+    train.coverage = data.coverage[(valid_idx + test_idx):]
+    train.baitdata = data.baitdata[(valid_idx + test_idx):]
+    train.gc = data.gc[(valid_idx + test_idx):]
+
+    return train, valid, test
+
+
+def sample_from_bed(bed_dict):
+    contig_sizes = {key: len(bed_dict[key]) for key in bed_dict.keys()}
+    total_size = sum(contig_sizes.values())
+
+    contig_key = np.random.choice(bed_dict.keys(), 1, p=[x / total_size for x in contig_sizes.values()])[0]
+
+    lowers = bed_dict[contig_key][0]
+    uppers = bed_dict[contig_key][1]
+    coverage = bed_dict[contig_key][2]
+    gcs = bed_dict[contig_key][3]
+
+    idx = np.random.randint(len(lowers))
+    return contig_key, lowers[idx], uppers[idx], coverage[idx], gcs[idx]
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# ~~~~~~~ Metrics ~~~~~~~~~~~~~~~~~
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+def gme(y_true, y_pred):
+    """calculates the root (geometric) mean squared error of the values."""
+    return K.exp(K.mean(K.log(K.abs(np.divide(y_true + .001, y_pred + .001) - 1))))
```

Comment:
You could use K.epsilon() instead of the hardcoded .001, this value is initialized in your ~/.keras/keras.json file and settable via K.set_epsilon(1e-05).  Also I think the division operator `/` is appropriately overloaded to handle this without np.divide()